nil/internal/network/host.go

```diff
@@ -15,46 +17,52 @@ import (
 	"github.com/libp2p/go-libp2p/p2p/security/noise"
 	quic "github.com/libp2p/go-libp2p/p2p/transport/quic"
 	"github.com/libp2p/go-libp2p/p2p/transport/tcp"
+	"github.com/rs/zerolog"
 )
 
 type Host = host.Host
 
 var defaultGracePeriod = connmgr.WithGracePeriod(time.Minute)
 
-func getCommonOptions(ctx context.Context, privateKey libp2pcrypto.PrivKey) ([]libp2p.Option, error) {
-	cm, err := connmgr.NewConnManager(100, 400, defaultGracePeriod)
+func getCommonOptions(ctx context.Context, conf *Config) ([]libp2p.Option, zerolog.Logger, error) {
+	pid, err := peer.IDFromPublicKey(conf.PrivateKey.GetPublic())
 	if err != nil {
-		return nil, err
+		return nil, zerolog.Nop(), err
 	}
 
-	pid, err := peer.IDFromPublicKey(privateKey.GetPublic())
+	logger := internal.Logger.With().
+		Stringer(logging.FieldP2PIdentity, pid).
+		Logger()
+
+	cm, err := connection_manager.NewConnectionManagerWithPeerReputationTracking(
+		ctx, conf.ConnectionManagerConfig, logger, 100, 400, defaultGracePeriod)
```

Comment:
What do low and hight mean here?
Why did you choose 100 and 400 here?

Reply:
This is the transposed code, I didn't change the values

Reply:
It's worth to understand that they mean

Reply:
> // lo and hi are watermarks governing the number of connections that'll be maintained.
// When the peer count exceeds the 'high watermark', as many peers will be pruned (and
// their connections terminated) until 'low watermark' peers remain.